impl/src/commonMain/kotlin/org/jetbrains/kotlinx/dl/impl/inference/imagerecognition/ImageRecognitionModelBase.kt

```diff
@@ -5,17 +5,24 @@
 
 package org.jetbrains.kotlinx.dl.impl.inference.imagerecognition
 
+import org.jetbrains.kotlinx.dl.api.summary.ModelHubModelSummary
+import org.jetbrains.kotlinx.dl.api.summary.ModelWithSummary
 import org.jetbrains.kotlinx.dl.api.core.shape.TensorShape
 import org.jetbrains.kotlinx.dl.api.inference.InferenceModel
+import org.jetbrains.kotlinx.dl.api.inference.loaders.ModelType
 import org.jetbrains.kotlinx.dl.api.preprocessing.Operation
+import org.jetbrains.kotlinx.dl.api.summary.EmptySummary
+import org.jetbrains.kotlinx.dl.api.summary.ModelSummary
 
 /**
  * Base class for image classification models.
  * @property [internalModel] model used for prediction
+ * @property [type] Corresponding [ModelType]. Used for model summary. Should be null for custom models.
  */
 public abstract class ImageRecognitionModelBase<I>(
     protected val internalModel: InferenceModel,
-) : InferenceModel by internalModel {
+    protected val type: ModelType<*, *>? = null
```

Comment:
I'd rather added something more broad here. We only need this parameter for the summary, why don't we add something like `modelKindDescription: String` or just make `type` a `String`. This will allow users with custom models use this property as well.